definitions/npm/styled-components_v5.x.x/flow_v0.104.x-/styled-components_v5.x.x.js

```diff
@@ -415,38 +415,13 @@ declare module 'styled-components' {
     <V>(V) => StyledShorthandFactory<V>
   >;
 
-  declare type ExtractProps<O> = $Call<
-    <T>(StyledComponent<T, any, any>) => T,
-    O
-  >;
-  declare type ExtractTheme<O> = $Call<
-    <T>(StyledComponent<any, T, any>) => T,
-    O
-  >;
-  declare type ExtractInstance<O> = $Call<
-    <T>(StyledComponent<any, any, T>) => T,
-    O
-  >;
-
   declare interface Styled {
-    <
-      Comp: StyledComponent<P, T, I>,
-      Theme: ExtractTheme<Comp>,
-      +Instance: ExtractInstance<Comp>,
-      OwnProps = ExtractProps<Comp>
-    >(
+    <Comp: React$ComponentType<P>, Theme, OwnProps = React$ElementConfig<Comp>>(
```

Comment:
Same question here and line 516 (`OwnProps = React$ElementConfig<Comp>` to `OwnProps: React$ElementConfig<Comp>`)

Reply:
See previous response about `:` vs `=`, in addition to that, I am not 100% sure how these "typed" or narrowed generics semantics work for the props-related types. It doesn't seem like there is any benefit of using them like that in this context (unless I was doing something wrong)